docs/integrations/persisting-store-data.md

```diff
@@ -731,3 +731,32 @@ interface BearState {
     removeItem: (name) => localStorage.removeItem(name),
   },
 ```
+
+If writing serialisation and deserialisation code is tedious, you can use third-party libraries to serialise and deserialise different types of data.
+
+SuperJSON serialises data along with its type, allowing the data to be parsed back to its original type upon deserialisation
```

Comment:
It depends whether @dai-shi and other contributors want to encourage using another library for this use case in the official Zustand docs.

Reply:
Should we change it to how to use persist with a custom serialization library? Like "How can I use a custom storage engine"?
![스크린샷 2023-10-26 01-13-24](https://github.com/pmndrs/zustand/assets/74400147/1a85c002-a1c7-4c66-8b2e-3c94d8d79eae)

If you use a lot of variables like Map, Set, Date, etc. that JSON.stringify() doesn't support, it would be nice to have a way in the documentation to avoid writing your own serialization/deserialization code

No problem using other libraries (ex) [serialize-javascript](https://github.com/yahoo/serialize-javascript), [devalue](https://github.com/Rich-Harris/devalue) , ...etc)

Reply:
@cheatkey I'm down with the idea of having this under "How can I use a custom storage engine?".